sdcm/nemesis.py

```diff
@@ -3820,6 +3820,108 @@ def _shrink_cluster(self, rack=None):
         self.log.info("Cluster shrink finished. Current number of nodes %s", num_of_nodes)
         InfoEvent(message=f'Cluster shrink finished. Current number of nodes {num_of_nodes}').publish()
 
+    # TODO: add support for the 'LocalFileSystemKeyProviderFactory' and 'KmipKeyProviderFactory' key providers
+    # TODO: add encryption for a table with large partitions?
+
+    def disrupt_enable_disable_table_encryption_aws_kms_provider(self):
+        # TODO: add check for Scylla that it supports the feature
+        if self.cluster.params.get("cluster_backend") != "aws":
+            raise UnsupportedNemesis("This nemesis is supported only on the AWS cluster backend")
+        if not self.cluster.params.get("aws_instance_profile_name_db"):
+            raise UnsupportedNemesis("This nemesis is supported only when an instance profile is used")
+        # TODO: update the 'kms_host' parameter when  dynamic key creation gets implemented.
+        self._enable_disable_table_encryption(additional_scylla_encryption_options={
+            'key_provider': 'KmsKeyProviderFactory', 'kms_host': 'kms_test'})
+
+    def _enable_disable_table_encryption(self, additional_scylla_encryption_options=None):
+        if additional_scylla_encryption_options and additional_scylla_encryption_options.get(
```

Comment:
it would be nice to have here the link to docs  instructions about this procedure

Reply:
Official docs don't include the `KMS` part: https://opensource.docs.scylladb.com/stable/operating-scylla/security/encryption-at-rest.html